[PATCH] iptables: bug fix: was calloc'ing n^2*size when only needed n*size for lookup responses

From: Wes Campaigne <westacular>
Date: Thu, 17 Feb 2011 08:07:56 -0500

---
 iptables.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/iptables.c b/iptables.c
index 724acd5..6d36fb8 100644
--- a/iptables.c
+++ b/iptables.c
@@ -618,7 +618,7 @@ host_to_addr(const char *name, unsigned int *naddr)
 
 		while (host->h_addr_list[*naddr] != (char *) NULL)
 			(*naddr)++;
-		addr = fw_calloc(*naddr, sizeof(struct in_addr) * *naddr);
+		addr = fw_calloc(*naddr, sizeof(struct in_addr));
 		for (i = 0; i < *naddr; i++)
 			inaddrcpy(&(addr[i]),
 				  (struct in_addr *) host->h_addr_list[i]);
-- 
1.6.5.GIT
